xcode/Subconscious/Shared/Services/Noosphere/Sphere.swift

```diff
@@ -348,6 +354,26 @@ public final class Sphere: SphereProtocol {
         return changes.toStringArray()
     }
     
+    /// Load the sphere of a user by petname, they must already
+    /// be present in your address book
+    public func traverseByPetname(petname: String) throws -> Sphere? {
+        let identity = try self.getPetname(petname: petname)
+        
+        let sphere = try Noosphere.callWithError(
+            ns_sphere_traverse_by_petname,
+            noosphere.noosphere,
+            sphere,
+            petname
+        )
+        
+        guard let sphere = sphere else {
+            logger.warning("Failed to traverse sphere for \(petname)")
+            return nil
+        }
+        
+        return Sphere(noosphere: noosphere, sphere: sphere, identity: identity)
```

Comment:
Given that we get a `ns_sphere_t` back from this call we may as well use it to initialize our own `Sphere` and skip another call.